### PR TITLE
feat: disable edit/duplicate entry for repeat panel

### LINKED ISSUE #21142: repeat panel should not be editable
Accoarding to [doc](https://grafana.com/docs/grafana/latest/reference/templating/#repeating-panels) only first repeat panel should be edit. other repeat panel (which as repeatPanelId field) shoud not be edit, because it will be override after next dashboard rebuild.
But in UI, there is not any notification or anything else to forbid user to edit child repeat panel,
and no guide or highlight to lead user edit first panel.
some user which is not familar with grafana will edit later repeat panel and save, but after refresh page, their change lossing without any notification. this is make no sense

**Steps to reproduce**:

**Actual Result**:

**Expected Result**

**Relevant WCAG Criteria:** [#.#.# WCAG Criterion](link to https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0)

**Environment**:
- Grafana version: 6.5.0
- Data source type & version: prometheus
- User OS & Browser: windows chrome
- Others:

### PR DESCRIPTION
**Pull request by okhowang**
**What this PR does / why we need it**:

avoid incorrect modify repeated panel

**Which issue(s) this PR fixes**:

Partially `resolve` #21142

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by torkelo** [comment]
This code is sadly duplicated for react panels, so needs to be updated here as well: https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/utils/getPanelMenu.ts#L110

**Comment by okhowang** [comment]
react panel updated,
I also add keyboard shortcut checking

File: public/app/core/services/keybindingSrv.ts [hunk 1]
```diff
@@ -339,6 +340,14 @@ export class KeybindingSrv {
       window.location.href = window.location.href + newUrlParam;
     });
   }
+
+  static canEdit(dashboard: DashboardModel): boolean {
+    return (
+      dashboard.meta.canEdit &&
+      dashboard.meta.focusPanelId &&
+      !dashboard.panels.find(e => e.id === dashboard.meta.focusPanelId)?.repeatPanelId
```
Comment (torkelo) [inline public/app/core/services/keybindingSrv.ts]: can you use getPanelById on daashboard model?  and not use the ? makes it a bit hard to read. Mabe this should be a function on the the DashboardModel itself, canEditPanel(panelId)   can be reused from all locations then

**Review by torkelo — CHANGES_REQUESTED** [review]

File: public/app/core/services/keybindingSrv.ts (hunk 1 above)
Comment (okhowang) [reply public/app/core/services/keybindingSrv.ts]: apologize for be unfamiliar with code.
I have added two method `canEditPanel` and `canEditPanelById` in DashboardModel.
and use them in all branch

**Comment by okhowang** [comment]
can we merge this pr?
any problem else?

**Review by torkelo — APPROVED** [review]
Looks great! Thanks for a bunch for this! ❤️

**Comment by torkelo** [comment]
Looks like there is a conflict in one file so can’t merge, if you can please try to merge with master and fix conflict otherwise I can do that next week

**Comment by okhowang** [comment]
conflict resolved

**Pull request merged**